### DISCUSSION TITLE
Material Design 3 in _tw

### DISCUSSION THREAD

**Discussion by rishiimmehra** [gregsullivan/_tw#105] [Q&A]
is there any possibility to include material design 3 in _tw theme if it's possible then how can I do it. Is there any documentation or guide to do it

**Comment by gregsullivan** [comment]
Hi Rishi—Thanks for posting your question!

I've never used Material Design in any of my projects. Could you give me a few more details on how you would integrate it with a static HTML page, just to give me some context? Are there Tailwind versions of Material Design components that you would generally use for a case like that one?

If there's a specific open-source project you're investigating with Tailwind-based Material Design components, knowing which one that is would help me give a more concrete answer to your question.

**Reply by rishiimmehra** [reply]
Actually I'm asking about m3 design and also known as material design 3 but when you ask is there tailwind based material design then I have little bit of research and I got into material tailwind
https://www.material-tailwind.com/
It have also available for html but I don't know how to integrate in _tw theme
Material Tailwind for HTML
https://www.material-tailwind.com/docs/html/installation

**Comment by rishiimmehra** [comment]
Working Scucessfully Material Tailwind I just follow the instruction and installed it sucessfully. The question i want to asked you to is what should i do with javascript should i import all componets javasript file into one js file or create each js file for each components

**Reply by gregsullivan** [reply]
Stepping back a bit here, how do you intend to integrate React with your WordPress theme? Unless I'm misunderstanding, you might be looking for something more in line with a headless WordPress experience.

The types of components that work with _tw are HTML-based, not React-based. Tailwind UI, TW Elements, etc. The components provide the HTML you would paste into your theme. There are still considerations in terms of setting up the JavaScript side, but not to the same extent as were you to attempt to add React components to your theme.

**Reply by rishiimmehra** [reply]
Material Tailwind is available for two framework first is html and second is react I jus installed html version

**Reply by gregsullivan** [reply]
My mistake, sorry about that!

I would import the JavaScript for all components into one file, personally. I only move away from that approach when using a single file is impractical because it gets too large.

**Comment by rishiimmehra** [comment]
Thanks Gregsullivan I think same as you.

**Reply by gregsullivan** [reply]
Awesome—good luck with your project!
